crates/runtime/src/component/mod.rs

```diff
@@ -68,121 +68,19 @@ pub enum ReplacedInstanceTarget {
-            | "wasi:http/incoming-handler@0.2.1"
-            | "wasi:http/incoming-handler@0.2.2"
-            | "wasi:http/outgoing-handler@0.2.0"
-            | "wasi:http/outgoing-handler@0.2.1"
-            | "wasi:http/outgoing-handler@0.2.2"
-            | "wasi:http/types@0.2.0"
-            | "wasi:http/types@0.2.1"
-            | "wasi:http/types@0.2.2"
-            | "wasi:io/error@0.2.0"
-            | "wasi:io/error@0.2.1"
-            | "wasi:io/error@0.2.2"
-            | "wasi:io/poll@0.2.0"
-            | "wasi:io/poll@0.2.1"
-            | "wasi:io/poll@0.2.2"
-            | "wasi:io/streams@0.2.0"
-            | "wasi:io/streams@0.2.1"
-            | "wasi:io/streams@0.2.2"
-            | "wasi:keyvalue/atomics@0.2.0-draft"
-            | "wasi:keyvalue/batch@0.2.0-draft"
-            | "wasi:keyvalue/store@0.2.0-draft"
-            | "wasi:logging/logging"
-            | "wasi:logging/logging@0.1.0-draft"
-            | "wasi:random/insecure-seed@0.2.0"
-            | "wasi:random/insecure-seed@0.2.1"
-            | "wasi:random/insecure-seed@0.2.2"
-            | "wasi:random/insecure@0.2.0"
-            | "wasi:random/insecure@0.2.1"
-            | "wasi:random/insecure@0.2.2"
-            | "wasi:random/random@0.2.0"
-            | "wasi:random/random@0.2.1"
-            | "wasi:random/random@0.2.2"
-            | "wasi:sockets/instance-network@0.2.0"
-            | "wasi:sockets/instance-network@0.2.1"
-            | "wasi:sockets/instance-network@0.2.2"
-            | "wasi:sockets/ip-name-lookup@0.2.0"
-            | "wasi:sockets/ip-name-lookup@0.2.1"
-            | "wasi:sockets/ip-name-lookup@0.2.2"
-            | "wasi:sockets/network@0.2.0"
-            | "wasi:sockets/network@0.2.1"
-            | "wasi:sockets/network@0.2.2"
-            | "wasi:sockets/tcp-create-socket@0.2.0"
-            | "wasi:sockets/tcp-create-socket@0.2.1"
-            | "wasi:sockets/tcp-create-socket@0.2.2"
-            | "wasi:sockets/tcp@0.2.0"
-            | "wasi:sockets/tcp@0.2.1"
-            | "wasi:sockets/tcp@0.2.2"
-            | "wasi:sockets/udp-create-socket@0.2.0"
-            | "wasi:sockets/udp-create-socket@0.2.1"
-            | "wasi:sockets/udp-create-socket@0.2.2"
-            | "wasi:sockets/udp@0.2.0"
-            | "wasi:sockets/udp@0.2.1"
-            | "wasi:sockets/udp@0.2.2"
-            | "wasmcloud:bus/lattice@1.0.0"
-            | "wasmcloud:bus/lattice@2.0.0"
-            | "wasmcloud:messaging/consumer@0.2.0"
-            | "wasmcloud:messaging/types@0.2.0"
-            | "wasmcloud:secrets/reveal@0.1.0-draft"
-            | "wasmcloud:secrets/store@0.1.0-draft" => continue,
-            _ => {}
-        }
-    };
+fn is_0_2(version: &str, min_patch: u64) -> bool {
+    if let Ok(semver::Version {
+        major,
+        minor,
+        patch,
+        pre,
+        build,
+    }) = version.parse()
+    {
+        major == 0 && minor == 2 && patch >= min_patch && pre.is_empty() && build.is_empty()
```

Comment:
Do we want to have the `pre.is_empty()` and `build.is_empty()` qualifiers here?

Put another way, would we expect for the following interfaces to work okay?
1. `wasi:http@0.2.5-pre.1`
1. `wasi:http@0.2.5+20250205`

Reply:
Such interfaces don't exist today in wasip2.
We can revisit this once/if that changes.

In any case we would need to follow what Wasmtime does here, since otherwise this would cause instantiation failures.